src/ui/common/src/components/pages/integrations/index.tsx

```diff
@@ -57,6 +78,26 @@ const IntegrationsPage: React.FC<Props> = ({
     forceLoad = true;
   }
 
+  // If the last storage migration failed, display the error message.
+  const { data, error, isLoading } = useStorageMigrationListQuery(
+      {
+        apiKey: user.apiKey,
+        limit: '1', // only fetch the latest result.
+      }
+  )
+  const lastMigration = data as StorageMigrationResponse[]
+  let lastFailedFormattedTimestamp: string | undefined = undefined;
+  if (lastMigration && lastMigration[0].execution_state.status === "failed") {
+    const date = new Date(lastMigration[0].execution_state.timestamps.registered_at);
+    const year = date.getFullYear();
+    const month = date.getMonth() + 1; // Note: getMonth() returns 0-based index
+    const day = date.getDate();
+    const hour = date.getHours() % 12; // Convert to 12-hour format
+    const minute = date.getMinutes();
+    const amPm = date.getHours() >= 12 ? "pm" : "am"; // Determine AM/PM based on hour
+    lastFailedFormattedTimestamp = `${year}-${month < 10 ? "0" : ""}${month}-${day < 10 ? "0" : ""}${day} ${hour}:${minute < 10 ? "0" : ""}${minute}${amPm}`;
```

Comment:
You should be able to use [`date.toLocaleString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString) to avoid having to do this by hand. Helps account for non-US date formats as well.